### ISSUE TITLE
version 1.9.1 on FreeBSD 10 clockid incorrect (fix included)

### ISSUE THREAD

**Issue by trushworth** [kivy/kivy#4086]
I've managed to get kivy 1.9.1 to build from source with python 3.4  and run on FreeBSD 10, but responses to input events were glacially slow.  The reason turns out to be in clock.py where the wrong value for _clockid was used.  A _clockid of 1 is CLOCK_VIRTUAL in FreeBSD.  I haven't tried the other two possible values (4,12) yet, I just picked the one that sounded closest to the Linux value and it seems to do the job.

```diff
--- clock.py-orig	2016-03-13 18:06:06.016779000 -0700
+++ clock.py	2016-03-13 18:08:20.648683000 -0700
@@ -261,6 +261,15 @@
 
                 if 'linux' in platform:
                     _clockid = 4  # CLOCK_MONOTONIC_RAW (Linux specific)
+                elif 'freebsd' in platform:
+                    # clockid constants from sys/time.h 
+                    # FreeBSD clock_gettime() uses a timespec of seconds and NANOseconds, not MICROseconds,
+                    # but the multiplier below is 1e-9 anyways, so the struct timeval is probably struct timespec
+                    # (which has a compatible memory layout with timeval) for linux and darwin as well.  Misleading, but no real harm.
+                    # _clockid = 4 # CLOCK_MONOTONIC (FreeBSD specific)
+                    _clockid = 11 # CLOCK_MONOTONIC_PRECISE (FreeBSD specific, known OK for 10.2)
+                    # _clockid = 12 # CLOCK_MONOTONIC_FAST (FreeBSD specific)
+                    Logger.debug('clock.py: {{{:s}}} clock ID {:d}'.format(platform,_clockid))
                 else:
                     _clockid = 1  # CLOCK_MONOTONIC
```

**Comment by akshayaurora** [comment]
@trushworth thanks for the fix. It would be better if you could just use git to include the changes.

If you have issues/confusion using git directly you could simply follow this method https://help.github.com/articles/editing-files-in-another-user-s-repository/ 

If you have further more complicated changes then you can::

1) Fork the repo https://github.com/kivy/kivy#fork-destination-box
2) Clone the forked repo locally  git clone https://github.com/trushworth/kivy
3) setup git https://help.github.com/articles/set-up-git/
3) make your edits into local clone.
4) git commit -a
5) git push to your repo and do a pr

**Comment by akshayaurora** [comment]
closed by 9736cb1f79b880d610f188cf0d6045fa868849fc
